lighthouse-core/config/budget.js

```diff
@@ -103,6 +103,8 @@ class Budget {
 
       if (b.resourceSizes !== undefined) {
         budget.resourceSizes = b.resourceSizes.map((r) => {
+          // Users supply budgets in KB but Lighthouse uses bytes throughout
+          r.budget = r.budget * 1024;
```

Comment:
my vote is to definitely not modify in place when it changes the unit (which is arguably a change in type) like this. The biggest issue is with this we can't always confidently interact with `budget`, since `budget.resourceSizes[0].budget` will be KB in the external file but bytes in the internal data structure, so in any given LH code we'll have to first figure out if we're in a context that's internal or external to LH before we can add or compare numbers.

I think better to stick to the (eventually :) well known external type and convert it in the few places we need to internally.

One other option is we could make a new property, `r.budgetBytes` or whatever, but not sure if that would be better.

> but Lighthouse uses bytes throughout

if it's any consolation, that's sadly not true :/ though we have tried to include a suffix when we can e.g. https://github.com/GoogleChrome/lighthouse/blob/d574377286133f82bfaad6623a61d07dd4aaecb4/lighthouse-core/config/constants.js#L22-L40